### PR TITLE
Show pending ordinal transactions

### PR DESCRIPTION
**Pull request by Imamah-Zafar**
# 🔘 PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# 📜 Background
Currently the pending status of ordinal transfer is not properly shown

Issue Link: #[[ENG-2618](https://linear.app/xverseapp/issue/ENG-2618/show-pending-ordinal-transaction)]

# 🔄 Changes
- The ui for the pending ordinal  transaction has been modified so that on the left side there would be the pending icon , then the title and on the right we wont be showing the amount in BTC anymore
- The `groupBtcTxsByDate` has been updated so pending transactions are always shown on the top of the list
- To be tested with  [core pr](https://github.com/secretkeylabs/xverse-core/pull/211) 

Impact:
- Explain the broader impact of these changes.
- How it improves performance, fixes bugs, adds functionality, etc.

# 🖼 Screenshot / 📹 Video

https://github.com/secretkeylabs/xverse-web-extension/assets/88320460/79b2828e-1ff0-47be-a787-bd8644a4dcaa

# ✅ Review checklist
Please ensure the following are true before merging:

- [x] Code Style is consistent with the project guidelines.
- [x] Code is readable and well-commented.
- [x] No unnecessary or debugging code has been added.
- [x] Security considerations have been taken into account.
- [x] The change has been manually tested and works as expected.
- [x] Breaking changes and their impacts have been considered and documented.
- [x] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/components/transactions/transactionTitle.tsx [hunk 1]
```diff
@@ -30,6 +30,18 @@ export default function TransactionTitle(props: TransactionTitleProps) {
     return tx.incoming ? t('TRANSACTION_RECEIVED') : t('TRANSACTION_SENT');
   };
 
+  const getBtcTokenTransferTitle = (tx: BtcTransactionData): string => {
+    if (tx.isOrdinal && tx.txStatus === 'pending') {
```
Comment (dhriaznov) [inline src/app/components/transactions/transactionTitle.tsx:34]: I think we can make the if-block with `tx.txStatus === 'pending'` check and the inner if-block for `tx.isOrdinal` check, should be easier to read

**Review by dhriaznov — APPROVED** [review]
LGTM, left a minor comment below

**Review by dhriaznov — DISMISSED** [review]

**Comment by DuskaT021** [comment]
@teebszet this pr is dependant on the pr formt the xverse-core pr https://github.com/secretkeylabs/xverse-core/pull/211 that is not included in the core version 1.6.2, nor previous ones

**Comment by DuskaT021** [comment]
@Imamah-Zafar I'll remove the label Release 0.17.0, cause it is dependent on a pr from core that is not included in the xverse-core version 1.6.2 and we want to have xverse-core version 1.6.2 in the release 0.17.0

**Comment by Imamah-Zafar** [comment]
Talked with @DuskaT021  and the latest version of core does have the relevant core PR for this to work hence this will be included in the Release 0.17.0

**Comment by DuskaT021** [comment]
@Imamah-Zafar needs updating the core version to 1.6.3

**Comment by dhriaznov** [comment]
As I can see, the core version is going to be upgraded when this PR https://github.com/secretkeylabs/xverse-web-extension/pull/576 is merged @DuskaT021

**Comment by DuskaT021** [comment]
> if it's tested, let's merge and label it with the next release @DuskaT021

Currently testing it, will be done in 30 mins 👍  @teebszet

**Comment by DuskaT021** [comment]
@Imamah-Zafar
Is this pr also covering the pending ordinals showing for ledger accounts as well?

**Comment by Imamah-Zafar** [comment]
> @Imamah-Zafar Is this pr also covering the pending ordinals showing for ledger accounts as well?

I didn't try the ledger flow myself but It should work for ledger as well since ultimately we use the same transaction history screens for all acounts

**Comment by DuskaT021** [comment]
Ordinal checklist:

| Regular accounts:| Ledger accounts: |
|--------|--------|
| - [ ] Ordinal (image, text, recursive)|- [ ] Ordinal (image, text, recursive)|
| - [ ] BRC-20 transfer (second step)| - [ ] BRC-20 transfer (second step)|

**Comment by DuskaT021** [comment]
left my comment in the linear ticket

**Comment by teebszet** [comment]
my bad: I wanted to include PR 211 in xverse-core@1.6.3, but in fact it wasn't. it will be included in 1.6.4 instead and we will need to bump the core version again here

File: src/app/screens/coinDashboard/transactionsHistoryList.tsx [hunk 2]
```diff
@@ -100,8 +100,10 @@ const groupBtcTxsByDate = (
     if (transaction.txStatus === 'pending') {
       pendingTransactions.push(transaction);
     } else {
-      processedTransactions[txDate] = [transaction] || [];
-      processedTransactions[txDate].push(transaction);
+      !processedTransactions[txDate]
+        ? (processedTransactions[txDate] = [transaction] || [])
```
Comment (dhriaznov) [inline src/app/screens/coinDashboard/transactionsHistoryList.tsx:104]: Even when the `transaction` is `undefined` the fallback value won't be used here as the first value will still be an array, so should we remove the `|| []` here?

Comment (Imamah-Zafar) [reply src/app/screens/coinDashboard/transactionsHistoryList.tsx]: yes makes sense I updated it. 👍

**Review by dhriaznov — APPROVED** [review]

**Comment by DuskaT021** [comment]
this is tested and it can be merged

**Comment by DuskaT021** [comment]
this is tested and it can be merged

**Pull request merged**